py-polars/polars/datatypes/classes.py

```diff
@@ -579,7 +579,15 @@ def __hash__(self) -> int:
 
     def __repr__(self) -> str:
         class_name = self.__class__.__name__
-        return f"{class_name}(categories={self.categories!r})"
+        if len(categories := self.categories) <= 6:
+            categories = [",".join(f"{cat!r}" for cat in categories)]
+        else:
+            categories = (
+                [",".join(f"{cat!r}" for cat in categories[:3])]
+                + ["…"]
+                + [",".join(f"{cat!r}" for cat in categories[-3:])]
+            )
```

Comment:
This is equivalent and easier to read/more efficient:

```suggestion
            categories = [
                ",".join(repr(cat) for cat in categories[:3]),
                "…",
                ",".join(repr(cat) for cat in categories[-3:]),
            ]
```

Reply:
The `repr` call is actually _less_ efficient that `f-strings` and the final result of those two expressions isn't the same  ;)

Before:
`["'c0','c1','c2'…'c3','c4','c5'"]`
After:
`' c 0 ' , ' c 1 ' , ' c 2 ' , ' c 3 ' , ' c 4 ' , ' c 5 '`